src/test/database/fakeListRemote.spec.ts

```diff
@@ -0,0 +1,103 @@
+import * as pathLib from "path";
+import { expect } from "chai";
+
+import { ListRemote } from "../../database/listRemote";
+
+export class FakeListRemote implements ListRemote {
+  data: any;
+  delay: number;
+
+  /**
+   * @constructor
+   * @param data           the fake database structure. Each leaf is an integer representing the subtree's size.
+   * @param largeThreshold the threshold to determine if a delete exceeds the writeSizeLimit.
+   *                       If the sum of all leaves to delete is larger than largeThreshold,
+   *                       the delete will return false.
+   */
+  constructor(data: any) {
+    this.data = data;
+    this.delay = 0;
+  }
+
+  async listPath(
+    path: string,
+    numChildren: number,
+    startAfter?: string,
+    timeout?: number
+  ): Promise<string[]> {
+    const startTime = Date.now();
+    await new Promise((resolve, _) => setTimeout(resolve, this.delay));
+    const latency = Date.now() - startTime;
+    if (timeout && latency >= timeout) {
+      return Promise.reject(new Error("timeout"));
+    }
+    const d = this._dataAtpath(path);
+    if (d) {
+      let keys = Object.keys(d);
+      /*
+       * We mirror a critical implementation detail of here. Namely, the
+       * `startAfter` option (if it exists) is applied to the resulting key set
+       * before the `limitToFirst` option.
+       */
+      if (startAfter) {
+        keys = keys.filter((key) => key > startAfter);
+      }
+      keys = keys.slice(0, numChildren);
+      return Promise.resolve(keys);
+    }
+    return Promise.resolve([]);
+  }
+
+  private _size(data: any): number {
```

Comment:
Prefixing a private function with an underscore is redundant

Reply:
Done.